### ISSUE TITLE
Fix FP S1128 (`unused-import`): Ignore imported symbols used as Vue.js directives

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#3790]
In the context of Vue.js Single File Components, the rule S1128 (`unused-import`) ignores imported symbols that are used as tags in the `template` section of such components. However, the rule is currently reporting false positives when imported symbols are used as tag attributes in the very same section.

```vue
<script lang="ts" setup>
  import { FooBar } from "./FooBar.vue"; // FP - FooBar is incorrectly flagged as unused
</script>
<template>
  <input v-foo-bar />
</template>
```

Reported in https://community.sonarsource.com/t/s1128-fp-with-use-vue3-directive-in-composition-api/86079

**Comment by aKzenT** [comment]
@yassin-kammoun-sonarsource I believe this exception has been implemented incorrectly and thus does not work to fix the false positives.

Please take a look at the official documentation: https://vuejs.org/guide/reusability/custom-directives#introduction A variable or import starting with "v" can be used as a directive. So vFocus becomes v-focus. However as it was implemented here, the v-Prefix is not necessary and in fact would still trigger the false positive (i.e. your rule is expecting "focus" as a valid import for "v-focus", but actually it should be "vFocus".

Please reopen and adjust.

**Comment by yassin-kammoun-sonarsource** [comment]
@aKzenT Duly noted, thanks!

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-97
